src/element/element_stream_writer.rs

```diff
@@ -111,7 +111,9 @@ where
                 if let Some(field_name) = field_name {
                     fields.push((field_name, element));
                 } else {
-                    return illegal_operation("Values inside a struct must have a field name.");
+                    return IonResult::illegal_operation(
+                        "Values inside a struct must have a field name.",
+                    );
```

Comment:
Generally speaking an error message without context is less helpful than one that does have context. It might be helpful to provide some indication here of which value does not have a field name. I'm not suggesting that we fix it in this PR, but what sort of additional book-keeping or capabilities might we need to provide context in cases like this?

I created https://github.com/amazon-ion/ion-rust/issues/590 to track/discuss. 

Perhaps it would be good to link to #590 in places like this, so that it's easier to come back to later?

On the other hand, if we *had* a more universal context description ability then we'd likely do some general pass when the capability was added.

Reply:
Agreed on all counts. (This particular case is a bit of an outlier--it shouldn't happen since it's serializing a fully-formed `Element`. Not setting a field name would be user error when driving an `IonWriter`.)